apps/jobs/views.py

```diff
@@ -228,3 +231,65 @@ def leaderboard(request, challenge_phase_split_id):
     paginator, result_page = paginated_queryset(distinct_sorted_leaderboard_data, request)
     response_data = result_page
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_remaining_submissions(request, challenge_phase_pk, challenge_pk):
+    challenge_model = get_model_object(Challenge)
+    challenge_model(challenge_pk)
+
+    challenge_phase_model = get_model_object(ChallengePhase)
+    challenge_phase = challenge_phase_model(challenge_phase_pk)
+
+    participant_team_pk = get_participant_team_id_of_user_for_a_challenge(
+        request.user, challenge_pk)
+
+    try:
+        ParticipantTeam.objects.get(pk=participant_team_pk)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {'error': 'You haven\'t participated in the challenge'}
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    max_submission_per_day = challenge_phase.max_submissions_per_day
+
+    max_submission = challenge_phase.max_submissions
+
+    submissions_done_today_count = Submission.objects.filter(
+        challenge_phase__challenge=challenge_pk,
+        participant_team=participant_team_pk,
+        challenge_phase=challenge_phase_pk,
+        submitted_at__gte=timezone.now().date()).count()
+
+    failed_submissions_count = Submission.objects.filter(
+        challenge_phase=challenge_phase_pk,
+        participant_team=participant_team_pk,
+        challenge_phase__challenge=challenge_pk,
+        status=Submission.FAILED,
+        submitted_at__gte=timezone.now().date()).count()
+
+    if ((submissions_done_today_count - failed_submissions_count) >= max_submission_per_day
+            or (max_submission_per_day == 0)):
+        date_time_now = timezone.now()
```

Comment:
Yes, In this part I am calculating the remaining time of the participant when the limit has been exhausted. It is done in the following way:
1. If the submissions that are done today - failed submissions is greater than the given maximum submissions per day then
2. variable date_time_now will take the utc time of that instant when this condition is true.
3. Now , I am calculating the midnight time & then subtracting the date_time_now from the midnight time to get the remaining time.
Please check if something is still not clear.